### DISCUSSION TITLE
How to prevent TextBox from updating TextProperty while typing?

### DISCUSSION THREAD

**Discussion by walbarm** [AvaloniaUI/Avalonia#8931] [Q&A] [answered]
I need the TextBox to not update the ViewModel while typing, but commit text change on Enter key press or on lose focus. I tried to subclass the TextBox but there are no overrides that I can use to achieve this functionality. Is there anyway to achieve that?

**Comment by timunie** [comment, marked as answer]
I think atm the only way is to bind the Text oneway and update the viewmodel manually on enter or on lost focus.

**Reply by timunie** [reply]
If you want this behavior for mire than one textbox you may want to create a behavior to do it.

**Reply by walbarm** [reply]
Thank you @timunie for answering my question. I actually need this as the default behavior for all TextBoxes on an application. I'm not familiar with behaviors so I need to learn about it. If there is a sample code specifically about controlling what value get set to the ViewModel by a behavior that you can point me to is appreciated. Thanks!

**Reply by timunie** [reply]
I hope one of those samples may help you: https://github.com/wieslawsoltes/AvaloniaBehaviors/tree/master/samples/BehaviorsTestApplication.Base/Views/Pages

**Reply by walbarm** [reply]
Thank you again @timunie for your answer. I looked into Avalonia Behaviors but couldn't find one that controls bindings like TextProperty bindings. My approach is to subclass TextBox and add another text property that get updated on Enter key or lose focus and bind the view model to the the new property. The problem is that NumericUpDown control subscribes to the Text property from code and I cannot override that from the NumericUpDown styles to tell what property to use for the text! So I have to override the TextBox text property in some way.
So my next question, is there a way to get binding subscriptions for a control and modify or replace it with a different one programmatically? Something similar to the WPF BindingOperations.GetBindingExpression(...)?

**Reply by timunie** [reply]
It's planned to implement some day. I don't know a due date: https://github.com/AvaloniaUI/Avalonia/issues/3754

But I found a behavior impl. which is already there: https://github.com/AvaloniaUI/Avalonia/issues/6071#issuecomment-861574988

**Reply by walbarm** [reply]
That's great @timunie! The behavior impl in https://github.com/AvaloniaUI/Avalonia/issues/6071#issuecomment-861574988 worked for me. Thank you.
